src/background_workers.h

```diff
@@ -1,3 +1,4 @@
+#define _GNU_SOURCE
```

Comment:
Why this? I believe this should be only set `#ifdef __linux__`

Reply:
#define _GNU_SOURCE is needed for the resolution of the following warning:
`implicit declaration of function ‘pthread_setname_np’ [-Wimplicit-function-declaration]`
does it break mac builds?

Reply:
changed as per our conversation on the new commit